Update to use Ghost’s user-editable navigation

backspace opened the pull request:
This closes #3. It converts the theme to use Ghost’s built-in navigation features rather than hard-coding links to pages. That means it’ll be easier for non-computer nerds to keep the site updated 🥳

You can [check it out](https://ghost-theme-sandbox.herokuapp.com/) on the theme “sandbox” site. It has different navigation than the true site to highlight this difference.

backspace commented:
eep I just noticed there’s a homepage header background that shows up after scrolling a bit on mobile, I’m not sure this update preserves that… and also the sandbox site has lost its theme (probably when the Heroku dyno went to sleep), so that setup needs some refining 😯

I’ll follow up re the header background in the morning 🤞🏻

backspace commented:
ah it does still show up! I’m not sure how exactly and maybe it would benefit from alignment with these changes but I think it’s fine for now at least haha

re the sandbox problem, my true dream is to create a Heroku deployment for every pull request, with something like [this approach](https://github.com/SNathJr/ghost-on-heroku/issues/9#issuecomment-633959179) to include the theme in the Git commit vs uploading a `.zip` that doesn’t persist on the ephemeral filesystem. Maybe I can get to this some time in the next week.

farfamahar commented:
How does the blog page look for you? I downloaded and opened this branch locally and the header background doesn't show up on the blog page for me. I think it might have to do with the way I set up [routes](https://ghost.org/docs/api/v3/handlebars-themes/routing/) to the blog page. Mine looks like this https://pastebin.com/nu6J2PLQ

Only way it works on my end is if I add

   `background-color: rgba(34,34,34,0.9);
    box-shadow: 0 19px 38px rgba(0,0,0,0.30), 0 15px 12px rgba(0,0,0,0.22);
    margin-bottom: 50px;`

to the .header class.

backspace commented:
Thanks for catching that! In `ba94237` I added a bit of ugliness that appends a `blog-template` class to `body` so it can get the same header that the other pages that need it have. I also added it for `post-template` which is the class individual blog posts get.

It’s visible in the [sandbox](https://ghost-theme-sandbox.herokuapp.com/blog/)… for now 😆

backspace commented:
I achieved my dream of a preview deployment for every PR! It required an unpleasant workaround because of how redirects and routes in the filesystem vs the database and don’t survive on Heroku’s ephemeral filesystem: the PR has a `.content-overrides/settings/routes.yaml` file that is copied to the deployment. It may be that we can actually keep that directory in `main` though… maybe that makes sense 🤔

But now you can see [on the preview](https://ghostprev-nav.herokuapp.com/blog/) that the blog has the proper header:

![image](https://user-images.githubusercontent.com/43280/91595183-baca3880-e928-11ea-8813-983761880ac9.png)

backspace commented:
If you think this is ready for merging, @FarfamaH, you could “approve” the pull request 🤓

![image](https://user-images.githubusercontent.com/43280/92018439-c8adfe00-ed1a-11ea-819c-0eab5342fcca.png)

If there are any conflicts with your work on #4, let me know if you want to pair on resolving them.
